apps/api/src/routers/user.py

```diff
@@ -0,0 +1,54 @@
+from logging import getLogger
+from typing import Annotated, Union
+
+from fastapi import APIRouter, Depends, Form, status
+from fastapi.responses import RedirectResponse
+from pydantic import BaseModel, EmailStr
+
+from auth import user_identity
+from auth.user_identity import User, use_user_identity
+from services import mongodb_handler
+from services.mongodb_handler import Collection
+from utils.user_record import Role
+
+log = getLogger(__name__)
+
+router = APIRouter()
+
+
+class IdentityResponse(BaseModel):
+    uid: Union[str, None] = None
+    status: Union[str, None] = None
+    role: Union[Role, None] = None
+
+
+@router.post("/login")
+async def login(email: EmailStr = Form()) -> RedirectResponse:
+    log.info("%s requested to log in", email)
+    if user_identity.uci_email(email):
+        # redirect user to UCI SSO login endpoint, changing to GET method
+        return RedirectResponse("/api/saml/login", status.HTTP_303_SEE_OTHER)
+    return RedirectResponse("/api/guest/login", status.HTTP_307_TEMPORARY_REDIRECT)
+
+
+@router.get("/logout")
+async def log_out() -> RedirectResponse:
```

Comment:
Nitpick: One word `logout` to be consistent with the route name?

Reply:
I initially chose two words to make the function name a verb, but sure, consistency might be better here.

Reply:
Changed in 7491f2e.